modules/substrate/src/verifier.rs

```diff
@@ -303,6 +307,77 @@ where
 
 		Ok(())
 	}
+
+	/// Import a finalized header without checking if this is true.
```

Comment:
What do you think about extracting this function out of `Verifier` struct, because it (as the name and docs suggest) does not verify that much (only handles authority set hand off).
I'm thinking about something like (on the call site):
```
InjectBestFinalizedHeader { storage } ::inject(header)
```

This will also draw a clear line between the old interface (and `Verifier`) and the new "unsafe" one.

Reply:
Given that we call `import_header` (new one) in a loop, perhaps from the API perspective it makes more sense to have something like:
```
AppendBestFinalized { storage }::append_finalized_chain(headers: &[Header])
```

Reply:
I think both of these suggestions make sense